### PR TITLE
Use DTO instead of entities for controllers input/ouput

### PR DESCRIPTION
**Pull request by mojiweb**
Use DTO instead of entities for controllers input/ouput
Update swagger definition

### CODE REVIEW

File: src/modules/matches/dto/match.dto.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+export class MatchDto {
+  @ApiProperty({ type: String })
+  id: string;
+
+  @ApiProperty({ type: TeamDto })
+  home_team: TeamDto;
+
+  @ApiProperty({ type: Number })
+  home_score: number;
```
Comment (Nas84) [inline src/modules/matches/dto/match.dto.ts:13]: **💭 thought:** ‏Do you think it is relevant to add a minimun properties into the ApiProperty decorator ?

File: src/modules/matches/dto/match.dto.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  @ApiProperty({ type: TeamDto })
+  away_team: TeamDto;
+
+  @ApiProperty({ type: Number })
+  away_score: number;
```
Comment (Nas84) [inline src/modules/matches/dto/match.dto.ts:19]: same comment as above

File: src/modules/matches/matches.service.ts [hunk 3]
```diff
@@ -2,30 +2,61 @@ import { Injectable } from '@nestjs/common';
...
+    match.away_team = await this.teamsRepository.findOneOrFail({ where: { id: matchData.away_team } });
+    match.away_score = 0;
+
+    const savedMatch = await this.matchesRepository.save(match);
+
+    return await MatchDto.fromEntity(savedMatch);
   }
 
-  async update(id: string, match: Match): Promise<UpdateResult> {
+  async update(id: string, matchData: UpdateMatchDto): Promise<UpdateResult> {
```
Comment (Nas84) [inline src/modules/matches/matches.service.ts:53]: **🤓 nitpick:** ‏i was able to update a match with negative value 😉

**Review by Nas84 — APPROVED** [review]
Overall looks good just some minor feedbacks

File: src/modules/matches/matches.service.ts (hunk 3 above)
Comment (mojiweb) [reply src/modules/matches/matches.service.ts:53]: well spotted, I'll do a dedicated MR to implement data validation

**Pull request merged**